### DISCUSSION TITLE
Best way to insert custom nodes into the editor

### DISCUSSION THREAD

**Discussion by dkrusenstrahle** [ueberdosis/tiptap#1258] [General Questions]
Hello,

I am building a Tiptap 2.0 editor.
In this editor I have created a custom node called blockLink (<block-link></block-link>).
How do I insert instances of this into the editor?

This kind of works but not sure if it is correct way?

`this.editor.chain().focus().setBlockLink().run();`

The SetBlockLink method looks like this and I have no idea what I am doing:

```
addCommands() {
      return {
        setBlockLink: ({ title = '', description = '', image = '', url = '' } = {}) => ({ tr, dispatch }) => {
          const { selection } = tr
          const node = this.type.create({title: title, url: url});
          if (dispatch) {
            tr.replaceRangeWith(selection.from, selection.to, node)
          }
          return true
        }
      }
    },
```

**Comment by philippkuehn** [comment]
It depends. You can use [insertContent](https://www.tiptap.dev/api/commands/insert-content) or [setNode](https://www.tiptap.dev/api/commands/set-node) or [replaceRange](https://www.tiptap.dev/api/commands/replace-range). Maybe it is best to look directly [into the code](https://github.com/ueberdosis/tiptap/tree/main/packages/core/src/commands) until the documentation is complete.

**Reply by dkrusenstrahle** [reply]
Thank you!
